helix-term/src/application.rs

```diff
@@ -471,7 +473,20 @@ impl Application {
                 }
             }
             signal::SIGCONT => {
-                self.claim_term().await.unwrap();
+                // Copy/Paste from same issue from neovim:
+                // https://github.com/neovim/neovim/issues/12322
+                // https://github.com/neovim/neovim/pull/13084
+                let mut retry_count = 10;
+                while retry_count > 0 {
+                    if self.claim_term().await.is_ok() {
+                        break;
+                    }
+                    retry_count -= 1;
+                }
+                if retry_count == 0 {
+                    panic!("couldn't set terminal raw mode");
```

Comment:
It could be useful to show the Err from claim_term. Also, claim_term can fail in other ways than not being able to enter raw mode so the message here should be a little more generic:

```rust
match self.claim_term().await {
    Ok(()) => break,
    Err(err) if retries == 10 => panic!("Failed to claim terminal: {}", err),
    Err(_) => continue,
}
```

I don't have strong opinions on `while` vs. `loop` vs. `for` here